### PR TITLE
Detect support for AVX/AVX2 in rocksdb

### PR DESCRIPTION
**Pull request by yamiM0NSTER**
Solves  #171. People with old hardware (pre-Haswell and old AMD CPU's) will have to compile source themselves or download special binaries for older architecture.

### CODE REVIEW

**Comment by yamiM0NSTER** [comment]
Please someone with old hardware test before PR is merged.

File: external/rocksdb/CMakeLists.txt [hunk 1]
```diff
@@ -1017,3 +1017,61 @@ option(WITH_TOOLS "build with tools" ON)
...
+          return -1;
+        }
+      }
+      return 0;
+    }"
+    HAVE_AVX2_EXTENSIONS)
+
+  # Set Flags
+  if(HAVE_AVX2_EXTENSIONS AND NOT MSVC_VERSION LESS 1800)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /arch:AVX2")
```
Comment (brandonlehmann) [inline external/rocksdb/CMakeLists.txt]: Can we added a message during the cmake via message(STATUS "RocksDB: AVX2 Enabled") / Disabled / Etc?

Comment (yamiM0NSTER) [reply external/rocksdb/CMakeLists.txt]: it shows HAVE_AVX_EXTENSIONS test and result of it, same for avx2 one

Comment (brandonlehmann) [reply external/rocksdb/CMakeLists.txt]: Yes, yes, see it was unclear in the test was for RocksDB or for TurtleCoin. If not, I'll deal.

**Comment by ZedPea** [comment]
How much speed do these instructions add?

I think it's pretty anti user to have them downloading a binary that doesn't work, and then they have to go into discord and ask what's up to find they have to download another version. Furthermore, if you're using a GUI wallet, you'll have to download a different version of walletd and drop it in, which is the opposite of easy.

If possible we should do run time checks for the instructions (is that possible? I think you were talking about it before) or just disable the AVX/AVX2 support. It's also not an easy one to download, most users know what version of windows they're using, but do they know what CPU features they have? Unlikely.

**Comment by yamiM0NSTER** [comment]
Uh as you might have seen the gains were noticable. The problem is with people running 6 generation old CPU's (pls bury the old hardware).
About removing AVX/AVX2 I proposed PORTABLE build for releases but was met with "make it smart".

**Comment by brandonlehmann** [comment]
@ZedPea  long term goal is to detect at runtime. The quick fix is to update the release files with ones built without AVX2 or AVX (I have a build right now that's okay). Further release builds will either (a) build w/o AVX2/AVX or; (b) have code that detects it at runtime.

**Pull request merged**